src/Functions/likePatternToRegexp.h

```diff
@@ -21,52 +21,84 @@ inline String likePatternToRegexp(std::string_view pattern)
     const char * const end = pattern.begin() + pattern.size();
 
     if (pos < end && *pos == '%')
-        ++pos;
+    {
+        while (++pos < end && *pos == '%')
+        {
+        }
+    }
     else
+    {
         res = "^";
-
-    while (pos < end)
+    }
+    while (pos != end)
     {
         switch (*pos)
         {
-            case '^': case '$': case '.': case '[': case '|': case '(': case ')': case '?': case '*': case '+': case '{':
+            case '^':
+            case '$':
+            case '.':
+            case '[':
+            case '|':
+            case '(':
+            case ')':
+            case '?':
+            case '*':
+            case '+':
+            case '{':
                 res += '\\';
                 res += *pos;
+                ++pos;
                 break;
             case '%':
-                if (pos + 1 != end)
+                ++pos;
+                if (pos != end)
```

Comment:
Braces around single-line if-statements and the like (e.g. l. 70) look quite unusual in the context of the ClickHouse codebase. The next person working on the code will likely remove them. Better don't even add them in the first place :)

Reply:
Interesting. 
As both @pufit and @rschu1ze requested for this, I think I'd better follow the current codebase's style as well